pkg/pluginmanager/manager.go

```diff
@@ -975,6 +975,13 @@ func doDeletePluginFromCatalog(pluginName string, target configtypes.Target, cat
 // If the central-repo is disabled, all discovered plugins will be installed.
 func SyncPlugins() error {
 	log.Info("Checking for required plugins...")
+	// ensure the k8s and tae current contexts are mutually exclusive.
```

Comment:
May be we can avoid this check in `SyncPlugins()` method by adding a check for ensuring mutual exclusion in root command's `PersistentPreRunE` hook.

Reply:
After discussion, removed this check and added the check in `PersistentPreRunE` hook.